src/rust/ide/src/model/module/registry.rs

```diff
@@ -0,0 +1,213 @@
+//! Module defining registry of module's states.
```

Comment:
I could, but this is an additional work which I don't know we use anywhere. When we will need such registry, I think it will be easy to split off then